pkg/storaged/content-views.jsx

```diff
@@ -0,0 +1,1713 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+(function() {
+    "use strict";
+
+    var cockpit = require("cockpit");
+    var dialog = require("./dialog");
+    var utils = require("./utils.js");
+
+    var permission = require("./permissions.js").permission;
+
+    var mustache = require("mustache");
+
+    var React = require("react");
+    var CockpitListing = require("cockpit-components-listing.jsx");
+    var Tooltip = require("cockpit-components-tooltip.jsx").Tooltip;
+
+    var _ = cockpit.gettext;
+    var C_ = cockpit.gettext;
+
+    function format_dialog (client, path, start, size, enable_dos_extended) {
```

Comment:
Does it make sense to have this format_dialog() code all in this same new file? Is it interrelated? If so, then go for it, but I thought it might make sense to have this in a 'format-dialog.js' file?